samples/net/sockets/echo_async/README.rst

```diff
@@ -0,0 +1,83 @@
+.. _async-sockets-echo-sample:
+
+Asynchronous Socket Echo Server
+###############################
+
+Overview
+********
+
+The sockets/echo-async sample application for Zephyr implements an
+asyncronous IPv4 TCP echo server using BSD Sockets like API with
+non-blocking sockets and ``poll()`` call. It is thus further extension
+of :ref:`sockets-echo-sample` sample. The purpose of this sample is
+still to show how it's possible to develop sockets application portable
+to both POSIX and Zephyr. As such, it is kept minimal and supports only
+IPv4 and TCP.
+
+The source code for this sample application can be found at:
+:file:`samples/net/sockets/echo_async`.
+
+Requirements
+************
+
+- :ref:`networking_with_qemu`
+- or, a board with hardware networking
+
+Building and Running
+********************
+
+Build Zephyr version of sockets/echo_async application like this:
+
+.. code-block:: console
+
+    $ cd $ZEPHYR_BASE/samples/net/sockets/echo_async
+    $ make pristine
+    $ make BOARD=<board_to_use>
+
+``board_to_use`` defaults to ``qemu_x86``. In this case, you can run the
+application in QEMU using ``make run``. If you used another BOARD, you
+will need to consult its documentation for application deployment
+instructions.
+
+After the start, the sample expects connections at 192.0.2.1, port 4242.
+The easiest way to connect is:
+
+.. code-block:: console
+
+    $ telnet 192.0.2.1 4242
+
+After connection, the application will echo back any line sent to it.
+Unlike the above-mentioned :ref:`sockets-echo-sample`, this application
+supports multiple concurrent client connections. So, you can open
+another terminal window and run the same telnet command as above.
+Maximum number of clients supported is 3, but can be easily adjusted
```

Comment:
The sample supports up to three connected clients, but this can be adjusted by changing ``NUM_FDS`` defined in the source code.